plugin/methodWrappers.go

```diff
@@ -129,14 +129,41 @@ func List(ctx context.Context, p Parent) (*EntryMap, error) {
 	return cachedList(ctx, p)
 }
 
-// Open reads the entry's content. Note that Open's results could be cached. Thus, when
-// using the reader returned by this method, use idempotent read operations such as ReadAt
-// or wrap it in a SectionReader. Using Read operations on the cached reader will change it
-// and make subsequent uses of the cached reader invalid.
+// Read reads up to size bits of the entry's content starting at the given offset.
+// It will panic if the entry does not support the read action. Callers can use
+// len(data) to check the amount of data that was actually read.
 //
-// TODO: Could we change this to Read? E.g. plugin.Read.
-func Open(ctx context.Context, r Readable) (SizedReader, error) {
-	return cachedOpen(ctx, r)
+// If the offset is >= to the available content size, then data != nil, len(data) == 0,
+// and err == io.EOF. Otherwise if len(data) < size, then err == io.EOF.
+//
+// Note that Read is thread-safe.
+func Read(ctx context.Context, e Entry, size int64, offset int64) (data []byte, err error) {
+	if !ReadAction().IsSupportedOn(e) {
+		panic("plugin.Read called on a non-readable entry")
+	}
+	data = []byte{}
+	if attr := e.attributes(); attr.HasSize() {
```

Comment:
It feels weird for a BlockReadable entry to have an unknown size attribute (why would the API let you block-read its content without telling you how big that content is?). Should we panic if we run into that case?

Reason I bring it up here is b/c if we do run into that kind of entry, then the plugin API's responsible for implementing `Read`'s semantics. `Readable` entries w/ the default signature don't need to worry about this because the `entryContentImpl` type takes care of all that.

Reply:
I'm also thinking about changing `size int64` => `size int`. Reason is b/c `len(data)` is an int.

Reply:
Is it worth returning `io.EOF` errors? When would it be useful for plugins to provide that information (aside from the block-readable entry w/ unknown size attribute case)? Asking b/c we ignore `io.EOF` in FUSE.

Reply:
I think leaving out `io.EOF` makes sense. I don't think we need it. We would potentially need it if we switched to `OpenDirectIO` mode (as noted in https://godoc.org/bazil.org/fuse/fs#HandleReader), which could help with #609.

Reply:
Oh, I also think we should require a BlockReadable entry to set the size attribute.